environment.yml

```diff
@@ -23,12 +23,13 @@ dependencies:
   - distributed
   - ecmwf-api-client
   - eofs
-  - esmpy >=8.6.0  # github.com/SciTools-incubator/iris-esmf-regrid/pull/342
+  - esmpy !=8.1.0,<8.6.0  # https://github.com/SciTools-incubator/iris-esmf-regrid/pull/342#issuecomment-2092921514
   - esmvalcore 2.11.*
   - fiona
   - fire
   - fsspec
-  - gdal
+  - gdal >=3.9.0
+  - importlib_metadata <8
```

Comment:
Could you add a link to an issue why this is needed?

Reply:
yes, the issue Emma stumbled upon last release, doing it now...

Reply:
done in 458189c5172d9b42563ffb03a1308d2c261cc6e7

Reply:
Following the trail of issues and pull requests from that link, it looks like importlib_metadata was required for Python 3.9 only, which we no longer support. Could we just take it out?

Reply:
I've seen it just recently with 3.10+ and esmpy==8.4.2 - let me just temporarily take it out see what happens in the tests

Reply:
as in: I've seen the esmpy metadata barf...

Reply:
Could you add a note to describe that, e.g. 'importlib_resources < 7 is required for ESMPy < 8.6 with Python 3.10 and 3.11' (if that's what it is)? Then, our future selves will at least have some clue why this pin is here and when we can remove it.

Reply:
added in cd3ab7eb9d30149cde8cfd8dec55e9a1cc76943b

Reply:
Thanks!